castanet/counter.py

```diff
@@ -0,0 +1,60 @@
+# Import necessary types for typing
+from typing import List, Tuple, Dict, Optional
+# Import LibCST
+import libcst as cst
+from castanet import generate_trees as generator
+
+import libcst.matchers as match
+
+# TODO: Create function definitions related to all necessary user stories (should look somewhat like match_imports)
+# TODO: Modify annotations to have node key as type and key as necessary information
+
+
+def match_imports(cast_dict):
+    """A function for counting the number of if statements in a Python program."""
+    imports_dictionary = {}
+    # Iterate through all of the Python files in a directory
+    for file in cast_dict:
+        # Find CASTs for each of these files
+        cast = cast_dict[file]
+        # Determine number of import statements for each file
+        imports = match.findall(cast, match.Import())
+        imports_dictionary[file] = len(imports)
+
+    return imports_dictionary
+
+    #matcher function to iterate through a file to get the total number of comments in a file
+def match_Comment(cast_dict):
+    """A function for counting the number of comments statements in a Python program."""
+    comment_dictionary = {}
+    # Iterate through all of the Python files in a directory
+    for file in cast_dict:
+        # Find CASTs for each of these files
+        cast = cast_dict[file]
+        # Determine number of comments for each file
+        Comment = match.findall(cast, match.Comment())
+        comment_dictionary[file] = len(Comment)
+
+    return Comment_dictionary
+
+#iterates through the number of comments found in each file and adds them up to get the total number of comments
+def total_comment(cast_dict):
```

Comment:
This function should be totally commented, including a properly formatted docstring. Comments describing the function should be _within_ the function and not outside, as it currently is (on line 40)

Reply:
Line 50 should be removed